### PR TITLE
JIRA: ABC-203 Format check

### PR DESCRIPTION
**Pull request by clusty**
This PR adds a test project that enforces that code formatting and API files are up to date.
This is ran on every PR.

### CODE REVIEW

File: TestProjects/APITests/Assets/Tests/APIScrapingTests.cs [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+                {
+                    filenames += $"{file}\n";
+                    shouldBe += "<<<<<<<<<<<<<<<<\n" + File.ReadAllText(file) + "\n<<<<<<<<<<<<<<<<<<<<<<";
+                }
+            }
+
+
+            Assert.IsTrue(allScraped,
+                "Some .api files have not been generated. Please make sure to run ApiScraping.Scrape() or configure the com.unity.coding package to your project in order to regenerate api files. Here are the api files that need to be generated\n" +
+                filenames + "ShouldBe:" + shouldBe);
```
Comment (judubu) [inline TestProjects/APITests/Assets/Tests/APIScrapingTests.cs:58]: That formatting looks suspicious. Won't you have a list of all the file names followed by ShouldBe: and a list of all the content of the files?

Comment (judubu) [inline TestProjects/APITests/Assets/Tests/APIScrapingTests.cs:58]: That formatting looks suspicious. Won't you have a list of all the file names followed by ShouldBe: and a list of all the content of the files?

File: TestProjects/APITests/Assets/Tests/APIScrapingTests.cs [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+            }
+        }
+
+        [TearDown]
+        public void TearDown()
+        {
+            for (var i = 0; i < m_ApiFileContents.Length; i++)
+            {
+                var apiAssetPath = AssetDatabase.GUIDToAssetPath(k_ApiFilesGUID[i]);
+                File.WriteAllText(apiAssetPath, m_ApiFileContents[i]);
```
Comment (judubu) [inline TestProjects/APITests/Assets/Tests/APIScrapingTests.cs:35]: Why are you rewriting the file?

File: TestProjects/APITests/Assets/Tests/APIScrapingTests.cs (hunk 1 above)
Comment (clusty) [reply TestProjects/APITests/Assets/Tests/APIScrapingTests.cs:58]: You would get a vomit of files that are not indented.

**Pull request merged**